src/call.c

```diff
@@ -783,7 +783,7 @@ static void call_decode_sip_autoanswer(struct call *call,
 }
 
 
-static int call_streams_alloc(struct call *call, bool got_offer)
+int call_streams_alloc(struct call *call, bool got_offer)
```

Comment:
Is `got_offer` arg needed here? Is there a difference with `call->got_offer`? At first look it looks like the same check.

Reply:
A cleanup might be possible.
`got_offer`:
- It is set `true`, if an SDP offer is received
- It is set to `false`, if an SDP answer is received

In `call_accept()` we simply could set the `call->got_offer` before the call to `call_streams_alloc()`.
In `sipsess_desc_handler()` we passed false. But even more correct it would be to simply leave `call->got_offer` untouched. See next commit!

Reply:
There is also the incoming OPTIONS. I think we may also set it in `call_alloc()`. Then we have the correct `call->got_offer` also in `ua_handle_options()`.

Reply:
The got_offer` refactoring looks great, less confusing now. Thanks.